src/yagolic.c

```diff
@@ -1,39 +1,31 @@
 #include "yagolic.h"
 
 int main() {
-  int rows, cols, i = 0, exit = 0;
+  int rows, cols, i = 0, end = 0;
   char sc = 1;
-  int *board;
 
-  system("clear");
-  printf("Welcome to this game of life! \n\n");
-  printf("How many ROWS shall your board have ?\n");
-  scanf("%d", &rows); // NOLINT acanf_s is not implemented in gnu c standard
-  printf("How many COLUMNS shall your board have ?\n");
-  scanf("%d", &cols); // NOLINT acanf_s is not implemented in gnu c standard
-  while ((sc = getchar()) != '\n' && sc != EOF)
-    ; /* discard the trailing '\n' character */
-  system("clear");
+  Board board = init_board();
 
-  board = calloc(rows * cols, sizeof *board);
-
-  random_init_board(board, rows, cols);
+  if (board.cells == NULL) {
+    fprintf(stderr, "Error: cells array is null\n");
+    exit(EXIT_FAILURE);
+  }
 
-  while (exit != 1) {
+  while (end != 1) {
     printf("This is your board after %d turns\n", i);
-    print_board(board, rows, cols);
+    print_board(board);
     printf("Please press ENTER to see the new generation or 'q' to quit.\n");
     while ((sc = getchar()) != '\n' && sc != EOF) {
       if (sc == 'q' || sc == 'Q') {
-        exit = 1;
+        end = 1;
       }
     }
     system("clear");
-    spread_life(board, rows, cols);
+    spread_life(board);
     i++;
   }
 
-  free(board);
+  free(board.cells);

```

Comment:
Haven't we merge MR that deletes dynamic memory allocation?

Reply:
No, I made changes that have not been approved yet.

Reply:
I really think it makes more sense to allocate from heap here since we can't know the size of the board before execution.

Reply:
and how heap allocation would help us?

Reply:
The number of cells required is not know before compilation. 

So it is either :

1. We allocate from stack based on the user input and we end up with a Variable-length array. I don't really want to use something that is banned from the Linux kernel itself.
2. We allocate from stack with a big static size. This is wasteful of memory, and it needs dedicated checks on the user inputs. It feels dirty.
3. We allocate from heap. We use just the right amount of memory but we have to handle the memory ourselves and allocation/de-allocation will be slower.

Unless I missed something, option 3 sounds the best to me.
What do you think?